### DISCUSSION TITLE
Redis cache expiration value

### DISCUSSION THREAD

**Discussion by savagedev** [ezeoleaf/larry#57] [Q&A] [answered]
Why is the Redis cache expiration periodicity value the same frequency as retrieving a random repo from Github? For example, running "make run-dev", the periodicity is one minute and this same value is used to expire the cached value. This means a randomly selected repo can be selected again too soon. Possibly this is by design? Is so then why? Otherwise it seems the cache expiration value should be a new config setting, independent of the run frequency.

**Comment by ezeoleaf** [comment, marked as answer]
Hi @savagedev,
That's a really good point, and also an important bug :|
Before we did a huge refactor for publishers and provider we were calculating the cache expiration value as cacheSize  * periodicity:
<img width="465" alt="image" src="https://user-images.githubusercontent.com/10358977/173294915-a6fa5210-79fc-4c56-9416-80a278c393ac.png">

But now, looks like that got lost in the refactor, so is something that we need to re add.  Will create a bug for this and feel free to work on it or I can pick it up later.

Thanks for finding this and reporting it :)
